lib/hanrahashiv.py

```diff
@@ -376,6 +375,164 @@ def ReplaceItemsInFormula(
     return AnimationGroup(*animations)
 
 
+def ModifyFormula(
+    formula : Tex or MathTex,
+    remove_items : list = [],
+    add_after_items : list =[], add_items_strs : list[list] =[], add_items_colors : list[list] = [],
+    replace_items : list[list] = [], replace_items_strs  : list[list] = [], replace_items_colors : list[list] = [],
+    new_formula_alignment=DL, add_items_animation_style = FadeIn, add_lag_ratio = 0.3
+):
+    """
+        All indices must be indices of original formula
+        In each list and sublist indices must be in increasing order
+    """
+    tex_string_list = [tex.get_tex_string() for tex in formula]
+    colors = [item.color for item in formula]
+
+    new_tex_string_list = tex_string_list.copy()
+    new_add_after_items = add_after_items.copy()
+    new_replace_items = []
+    for i in range(len(replace_items)):
+        new_replace_items.append(replace_items[i].copy())
+    new_colors = colors.copy()
+
+    if type(formula) == Tex:
+        for i in range(len(replace_items_strs)):
+            for j in range(len(replace_items_strs[i])):
+                if replace_items_strs[i][j][0] != '$':
+                    replace_items_strs[i][j] = '$' + replace_items_strs[i][j]
+                if replace_items_strs[i][j][-1] != '$':
+                    replace_items_strs[i][j] = replace_items_strs[i][j] + '$'
+        for i in range(len(add_items_strs)):
+            for j in range(len(add_items_strs[i])):
+                if add_items_strs[i][j][0] != '$':
+                    add_items_strs[i][j] = '$' + add_items_strs[i][j]
+                if add_items_strs[i][j][-1] != '$':
+                    add_items_strs[i][j] = add_items_strs[i][j] + '$'

```

Comment:
I would rather check with `startswith` and `endswith`, so that it didn't fail on empty strings.

Also, you do the same thing in two places. Create a small function inside `ModifyFormula` just for that purpose. variable names would become shorter as well :)